plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/internal/CloudSdkContextFunction.java

```diff
@@ -61,11 +61,9 @@ public Object compute(IEclipseContext context, String contextKey) {
 
     Object path = context.get(PreferenceConstants.CLOUDSDK_PATH);
     File location = toFile(path);
```

Comment:
I don't think location should be unused, I think there's value in being able to specify an alternative path this way — it certainly doesn't hurt.

The background here is that context functions are only re-evaluated (which we'll want to do if the preference changed) when some input used by the function is changed.  We trigger the re-evaluation here by retrieving the SDK Path from the referencing contexts, and then setting the SDK Path on those contexts when the preference changes (see further above to the `referencedContexts`, which is used by the `Activator`'s `IPreferenceChangeListener` and `CloudSdkContextFunction#sdkPathChanged()`).

That said, there's a logic error here in that a path value explicitly set in the context will be overwritten when the preference is next changed, and `compute()` should be:
```
    Object path = context.get(PreferenceConstants.CLOUDSDK_PATH);
    if(path == null) {
      // record this context as using the preference value
      referencedContexts.add(context);
    }
```

But if you strongly believe there should only be a single SDK Path, then we should use a different context key name than `PreferenceConstants.CLOUDSDK_PATH`, like `"sdkPathChanged"`.